### ISSUE TITLE
RGB Order is for QT board is no longer correct as of version 1.1.1

### ISSUE THREAD

**Issue by apendley** [adafruit/Adafruit_IS31FL3741#5]
- Arduino board:  `Adafruit ItsyBitsy nRF52840 Express`

- Arduino IDE version: `1.8.13`

- List the steps to reproduce the problem below (if possible attach a sketch or
  copy the sketch code in too): Between version 1.1.0 and version 1.1.1, the RGB order no longer appears to be correct for the `Adafruit_IS31FL3741_QT` class. The components now appear to be ordered "GBR" instead of the expected "RGB" when using the `color565(r, g, b)` method.

**Comment by PaintYourDragon** [comment]
Can’t reproduce. color565(r,g,b) is, in fact, coming up r,g,b on QT matrix. Do you have a minimal example of the behavior that can attach here?

**Comment by PaintYourDragon** [comment]
(QT matrix color order was _fixed_ in 1.1.1 — it’s possible you had a sketch with a workaround for <=1.1.0’s incorrect order)

**Comment by apendley** [comment]
This is very bizarre. I am reproducing this with both the `qtmatrix-text` and `qtmatrix-text-buffered` examples, with only one modification; I am changing:
```
uint32_t color888 = matrix.ColorHSV(65536 * i / strlen(text));
uint16_t color565 = matrix.color565(color888);
```
to
```
uint16_t color565 = matrix.color565(255, 0, 0);
```

The result is green text 🤔.

I reverted back to the version before the fix, and the text appears red again, as expected. I've tried this ItsyBitsy bluefruit, m0, m4, and rp2040 all with the same result.

**Comment by PaintYourDragon** [comment]
Indeed bizarre! Getting red text here.

@ladyada Has anything changed among different production runs of these matrices? I didn’t get in on the first batch. Mine are dated 2021-08-24 on baggie, silver Sharpie dot on the right STEMMA connector if that means anything.

**Comment by ladyada** [comment]
we did change LEDs, the very first batch had a dot on the left i think, then changed to dot on right. but i thought the LEDs were the same color order :/

**Comment by PaintYourDragon** [comment]
Uh oh. If you still have a “classic” matrix there, could you isolate it for me? We can probably fix this with an extra arg in the constructor or begin(), a la NeoPixel library. Would much prefer to have one in-hand for testing rather than flying on instruments though.

Is this only going to affect some early QT matrices, or do glasses, etc. also have the component change?

**Comment by apendley** [comment]
Oh no! I have 2 that I ordered from the very first batch I think? I have 2 more arriving in the post this evening, when I get them I'll test them and see if the result is different.

**Comment by apendley** [comment]
I just received my order in the mail. Both of the new ones have a silver sharpie on the right QT port, and I can confirm that the RGB order is indeed correct on these newer ones.

**Comment by PaintYourDragon** [comment]
Thanks for checking. I’ll add something to the constructor to take care of this, should then be possible to mix old & new. Might be a couple days due to other tasks.

**Comment by apendley** [comment]
Thank you! And no worries about the time frame, I have a local fix I can use in the meantime.

**Comment by PaintYourDragon** [comment]
I think the problem’s resolved, but before I merge this and potentially ruin everyone’s day…could you please download this version and give it a test on a “left dot” QT matrix? (I don’t have one here)
https://github.com/adafruit/Adafruit_IS31FL3741/tree/pb-rgbfix

The Adafruit_IS31FL3741_QT and Adafruit_IS31FL3741_QT_buffered constructors now accept an optional color-order argument. Default if unspecified is the newer “right dot” matrix, as that seems to be the majority of them right now. A set of valid colors are enumerated in the header file.

e.g. instead of:
`Adafruit_IS31FL3741_QT matrix;`

You can override color order with:
`Adafruit_IS31FL3741_QT matrix(IS3741_GBR);`

I actually don’t know the correct color order for left-dot matrices, so you may need to try a few permutations until things look right.

**Comment by apendley** [comment]
I should have some time to test this later today! I recently hacked apart the buffered text demo to allow me to use 2 unmodified matrices side by side on a QT Py RP2040 (1 per I2C port), I'll swap one out with my original one and try this out.

**Comment by apendley** [comment]
I was able to get the original matrix to use the correct color order with the new code by passing in `IS3741_RBG`. One interesting note is that there are considerable differences in the colors across the two versions (the new ones are much more vibrant!).

**Comment by ladyada** [comment]
yeah we had LEDs and then our supplier couldnt get us more, so we found another vendor - but the LEDs are slightly different :/

**Comment by apendley** [comment]
That's a bummer. Personally I don't mind. I can find a project for this little misfit 😄 Thanks for such a quick turnaround on the fix!

**Comment by ladyada** [comment]
‧͙⁺˚*･༓☾ silicon shortage  ☽༓･*˚⁺‧͙

**Comment by PaintYourDragon** [comment]
Thank you for testing! Just published 1.2.0 release; should be live in Arduino Library Manager once it trickles through all the gears.
